R/preprocess.R

```diff
@@ -233,6 +236,56 @@ enw_reporting_triangle_to_long <- function(obs) {
   return(reports_long[])
 }
 
+#' @title FUNCTION_TITLE
+#' @description FUNCTION_DESCRIPTION
+#' @param obs A `data.table` with the observations to complete
+#' 
+#' @param min_date The first date from which to complete.
+#' 
+#' @param max_date The last date until which to complete.
+#' 
+#' @param groups A `data.table` with one row per group, a column for each group
+#' attribute and a column `group` holding the group ID.
+#' 
+#' @param max_delay The maximum reporting delay.
+#' 
+#' @param include_missing Should entries for cases with missing reference date
+#' be completed as well?, Default: FALSE
+#' 
+#' @return A `data.table` with completed entries for all combinations of reference
+#' dates, groups and possible report dates.
+#' @family preprocess
+enw_complete_dates <- function(obs, min_date, max_date, groups, max_delay, include_missing = FALSE) {
```

Comment:
Just by the way: I believe the completion of the missing reporting dates can also be done using `dcast` in the case of the **reporting triangle** (and is actually already done). In other words, it is not necessary for the reporting triangle, but I think it is necessary for the other data structures. One option to think about would be to implement a more efficient version of the below function using dcast. Then again, this is more explicit and needs no reshaping of data structures...

Reply:
yup that was intentional as happens quite a bit in data I've seen but I missed the fact it wouldn't enter into the metadata and so won't be assigned features/will break (I think). 

This looks quite efficient whilst also being clear so I'd go with this implementation vs `dcast` magic.

Doing it all at once (and maybe before preprocessing makes lots of sense).